Font.js

```diff
@@ -119,40 +34,23 @@ class Font extends EventManager {
      * Just like Image and Audio, we kick everything off when
      * our `src` gets assigned.
      *
-     * @param {string} url
+     * @param {string} source url for this font ("real" or blob/base64)
      */
-    set src(url) {
-        this.__src = url;
-        this.defineFontFace(this.name, url, this.options);
-        this.loadFont(url);
-    }
-
-    /**
-     * This is a blocking operation.
-     */
-    defineFontFace(name, url, options) {
-        if (typeof document !== "undefined") {
-            let format = getFontCSSFormat(url);
-            if (!format) return;
-            let style = document.createElement(`style`);
-            style.className = `injected by Font.js`;
-            let rules = Object.keys(options).map(r => `${r}: ${options[r]};`).join(`\n\t`);
-            style.textContent = `
-    @font-face {
-        font-family: "${name}";
-        ${rules}
-        src: url("${url}") format("${format}");
-    }`;
-            this.styleElement = style;
-            document.head.appendChild(style);
-        }
+    set src(src) {
+        this.__src = src;
+        (async() => {
+            if (globalThis.document && !this.options.skipStyleSheet) {
+                await setupFontFace(this.name, src, this.options);
+            }
+            this.loadFont(src);
+        })();
```

Comment:
while the `src` assignment itself can only be synchronous, the actual loading should happen async.